From a84611b574271bf5b03ca9a909fcd09708ceb8fa Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Thu, 7 Apr 2016 18:05:08 +0800
Subject: [PATCH 1022/5242] MLK-12868-01 cpufreq: imx: get old_freq from
 policy->cur

commit  6b1d0c55e8494f813b584e58a7b6cd77a034247a from
https://source.codeaurora.org/external/imx/linux-imx.git

Get the old_freq from the policy->cur.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/cpufreq/imx6q-cpufreq.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/cpufreq/imx6q-cpufreq.c b/drivers/cpufreq/imx6q-cpufreq.c
index ec1964a..e7befaf 100644
--- a/drivers/cpufreq/imx6q-cpufreq.c
+++ b/drivers/cpufreq/imx6q-cpufreq.c
@@ -82,7 +82,7 @@ static int imx6q_set_target(struct cpufreq_policy *policy, unsigned int index)
 
 	new_freq = freq_table[index].frequency;
 	freq_hz = new_freq * 1000;
-	old_freq = clk_get_rate(clks[ARM].clk) / 1000;
+	old_freq = policy->cur;
 
 	opp = dev_pm_opp_find_freq_ceil(cpu_dev, &freq_hz);
 	if (IS_ERR(opp)) {
-- 
1.7.9.5

